controllers/machinedeletionremediation_controller_test.go

```diff
@@ -227,7 +243,21 @@ var _ = Describe("Machine Deletion Remediation CR", func() {
 				It("returns the same delete failure error", func() {
 					Eventually(func() bool {
 						return plogs.Contains(mockDeleteFailMessage)
-					}, 30*time.Second, 1*time.Second).Should(BeTrue())
+					}, "10s", "1s").Should(BeTrue())
+
+					verifyStatusCondition(v1alpha1.ProcessingConditionType, metav1.ConditionFalse)
+					verifyStatusCondition(v1alpha1.SucceededConditionType, metav1.ConditionTrue)
```

Comment:
machine deletion fails == succeeded? 🤔

Reply:
I see this is quite unclear, so I am opened to better ideas, however mock client fails deletion _only once_, then it succeeds at the next try

https://github.com/clobrano/machine-deletion-remediation/blob/99704c46dfd12f6219285c2e0796c6caffae4e6c/controllers/suite_test.go#L87-L91

Reply:
Maybe revert that "once only" change in the mock delete function, and reset `onDeleteError` in a `DeferCleanup()` in the `BeforeEach()?` block?

Reply:
The problem is that the `Reconcile` loop will keep retrying to delete the machine forever.
This is something I wanted to address with #80, however is not that easy to come up with a proper retry and IIUC it is not that possible that a `Deletion` request will get errors forever.

Reply:
> The problem is that the Reconcile loop will keep retrying to delete the machine forever.

Why is that a problem? You want to test the correct conditions while this happens, so here we are 🙂

Reply:
> forever

well, just until you reset onDeleteError after verifying the conditions

Reply:
Oh, I haven't considered that the test might have finished already at that point 👍